src/libsyntax_pos/symbol.rs

```diff
@@ -1011,7 +1011,15 @@ pub mod sym {
 }
 
 impl Symbol {
-    fn is_used_keyword_2018(self) -> bool {
+    pub fn is_used_keyword(self) -> bool {
+        self.is_used_keyword_2015() || self.is_used_keyword_2018()
+    }
+
+    pub fn is_used_keyword_2015(self) -> bool {
+        self >= kw::As && self <= kw::While
```

Comment:
Could you remove the `symbol.rs` changes?
`is_used_keyword` needs a span to work correctly.

If there's no available span, you can make a dummy spanned ident with `Ident::with_dummy_span` a the call point.

Reply:
done